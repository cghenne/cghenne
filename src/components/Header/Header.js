// @flow
import React from 'react';
import './Header.css';

const Header = () => {
  return (
    <header>
      <h1>
        <span className="secondary">Cathleen</span> Ghenne
      </h1>
      <h2>
        <span className="secondary">Front-End</span> Engineer
      </h2>
    </header>
  );
};

export default Header;
