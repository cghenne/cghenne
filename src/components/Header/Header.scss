@import 'styles/utils.scss';

header {
  text-align: center;
  color: $color-primary;
  font-family: $font-family-title !important;

  h1 {
    font-size: $font-size-xx-large;
    text-transform: uppercase;
    letter-spacing: 3px;
    margin-bottom: $spacing;

    @include mq($from: tablet) {
      font-size: $font-size-xxx-large;
    }
  }

  h2 {
    font-size: $font-size-regular;
    text-transform: uppercase;
    letter-spacing: 3px;
    margin-bottom: $spacing;

    @include mq($from: tablet) {
      font-size: $font-size-large;
    }
  }

  .secondary {
    color: $color-secondary;
  }
}
