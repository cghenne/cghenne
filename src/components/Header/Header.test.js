import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  const wrapper = shallow(<Header />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render header', () => {
    expect(wrapper.find('header').exists()).toEqual(true);
  });

  it('should render my name in h1', () => {
    expect(wrapper.find('h1').text()).toEqual('Cathleen Ghenne');
    expect(
      wrapper
        .find('.secondary')
        .first()
        .text()
    ).toEqual('Cathleen');
  });

  it('should render my profession in h2', () => {
    expect(wrapper.find('h2').text()).toEqual('JavaScript Engineer');
    expect(
      wrapper
        .find('.secondary')
        .last()
        .text()
    ).toEqual('JavaScript');
  });
});
