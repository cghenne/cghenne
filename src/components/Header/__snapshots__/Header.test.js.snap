// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without crashing 1`] = `
<header>
  <h1>
    <span
      className="secondary"
    >
      Cathleen
    </span>
     Ghenne
  </h1>
  <h2>
    <span
      className="secondary"
    >
      JavaScript
    </span>
     Engineer
  </h2>
</header>
`;
